src/main/java/com/transferwise/openbanking/client/api/payment/common/BasePaymentClient.java

```diff
@@ -27,9 +25,10 @@ protected String getClientCredentialsToken(AspspDetails aspspDetails) {
         return getAccessToken(getAccessTokenRequest, aspspDetails);
     }
 
-    protected String exchangeAuthorizationCode(String authorizationCode, AspspDetails aspspDetails) {
-        GetAccessTokenRequest getAccessTokenRequest = GetAccessTokenRequest.authorizationCodeRequest(authorizationCode,
-            tppConfiguration.getRedirectUrl());
+    protected String exchangeAuthorizationCode(AuthorizationContext authorizationContext, AspspDetails aspspDetails) {
```

Comment:
When exchanging the authorization code for an access token, the redirect URL value **must** be the same value as we specified in the ASPSP (payment) authorisation URL we redirected the user to. 

https://tools.ietf.org/html/rfc6749#section-4.1.3